crates/subspace-service/src/dsn/import_blocks/root_blocks.rs

```diff
@@ -97,36 +110,74 @@ impl RootBlockHandler {
                     match request_result {
                         Ok(RootBlockResponse { root_blocks }) => {
                             trace!(%peer_id, "Last root block request succeeded.");
+                            if root_blocks.iter().any(|rb| rb.is_some()) {
+                                participating_peers += 1;
+                            }
 
-                            let last_root_block = root_blocks
-                                .iter()
-                                .max_by(|rb1, rb2| compare_optional_root_blocks(rb1, rb2));
+                            let root_block_deduplication_map = BTreeMap::from_iter(
+                                root_blocks
+                                    .iter()
+                                    .filter_map(|rb| rb.map(|rb| (rb.hash(), rb))),
+                            );
+                            let sanitized_root_blocks = root_block_deduplication_map
+                            .values()
+                            .collect::<Vec<_>>();
```

Comment:
I'm not sure what you wanted to do here. Why do we need to de-duplicate it like this? Also this seems to indicate this is receiving a vector of options, but I don't understand why would last root blocks be optional. It should just return a vector of whatever length it can where all entries are present and are in canonical order (probably just a hash map).

Reply:
If for any reason request returns duplicated root blocks - it will spoil the root block voting. We can't guarantee a proper response, this is why I added sorting and deduplication when it matters.

 Also, I will probably remove options from the result for a missing root block for a segment index, it's confusing.

Reply:
> If for any reason request returns duplicated root blocks

Then response is invalid and peer should be banned for not following the protocol.